### ISSUE TITLE
`AnnotatedString` doesn't have an `IOBuffer` method

### ISSUE THREAD

**Issue by MasonProtter** [JuliaLang/StyledStrings.jl#112]
This caused some unexpected errors in https://github.com/timholy/ProgressMeter.jl/issues/343#issuecomment-2779803870. The way you're supposed to find the number of lines in a string is with `countlines(IOBuffer(str))` and this errors:
```julia
julia> str = Base.AnnotatedString(" julia", [(2:6, :face, :magenta)])
" julia"

julia> countlines(IOBuffer(str))
ERROR: MethodError: no method matching IOBuffer(::Base.AnnotatedString{String})
The type `IOBuffer` exists, but no method is defined for this combination of argument types when trying to construct it.

Closest candidates are:
  IOBuffer(::SubString{String})
   @ Base strings/io.jl:309
  IOBuffer(::String)
   @ Base strings/io.jl:308
  IOBuffer(; read, write, append, truncate, maxsize, sizehint)
   @ Base iobuffer.jl:119
  ...

Stacktrace:
 [1] top-level scope
   @ REPL[14]:1

```

**Comment by tecosaur** [comment]
This is easy enough to fix, should there be an `IOBuffer(::AbstractString)` method though?
